reservation-service/src/main/java/com/reservation/reservationservice/service/RequestService.java

```diff
@@ -1,7 +1,82 @@
 package com.reservation.reservationservice.service;
 
+import com.reservation.reservationservice.dtos.ReservationDTO;
+import com.reservation.reservationservice.dtos.TableRequestDTO;
+import com.reservation.reservationservice.exceptions.BadRequestException;
+import com.reservation.reservationservice.model.*;
+import com.reservation.reservationservice.model.enums.RequestStatus;
+import com.reservation.reservationservice.repository.AccomodationRepository;
+import com.reservation.reservationservice.repository.GuestRepository;
+import com.reservation.reservationservice.repository.RequestRepository;
+import com.reservation.reservationservice.repository.ReservationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 @Service
 public class RequestService {
+    @Autowired
+    private RequestRepository requestRepository;
+    @Autowired
+    private AccomodationRepository accomodationRepository;
+    @Autowired
+    private ReservationRepository reservationRepository;
+
+    @Autowired
+    private GuestRepository guestRepository;
+    public ReservationDTO approveRequest(String id) throws BadRequestException {
+
+        Optional<Request> request =  this.requestRepository.findById(id);
```

Comment:
Why not put request = this.requestRepository.findById(id).get() and maybe orElseThrow(EntityNotFound::new)

Reply:
done